lib/util.js

```diff
@@ -403,6 +403,108 @@ module.exports = {
     };
   },
 
+  /**
+   * Generates object containing objects of tags and their requests
+   * @param {object} spec - openAPI spec object
+   * @returns {object} returns an object containing objects of tags and their requests
+   */
+  /* eslint-disable max-depth */
+  generateFolderObjectFromTags: function(spec) {
+    var globalTags = spec.tags,
+      localTags = [],
+      globalTag = '',
+      paths = spec.paths,
+      currentPathObject = {},
+      operationObject = {},
+      folderObj = {
+        withTags: {},
+        withoutTags: {
+          requests: []
+        }
+      },
+      collectionVariables = {},
+      commonParams = '',
+      summary = '',
+      pathLevelServers = '',
+      name = '',
+      description = '';
+
+    // checking if tags are defined at the root level
+    if (globalTags.length > 0) {
+      for (let tag in globalTags) {
+        if (globalTags.hasOwnProperty(tag)) {
+          globalTag = globalTags[tag];
+          name = globalTag.name;
+          if (globalTag.hasOwnProperty('description')) {
```

Comment:
redundant
if the condition fails, `description` will still be undefined or null

Reply:
You sure?
because we've defined the description to an empty string.
See line 430